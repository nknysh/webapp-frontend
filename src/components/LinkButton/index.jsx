import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { theme } from 'styles';

// Why LinkButton?
// Semantically, a link takes the user somewhere else. A button performs
// an action in situ, or dispatch events. But since designers keep making
// buttons look like links, we need "LinkButton"!

export const LinkButton = props => {
  const { children, ...buttonProps } = props;
  return (
    <button {...buttonProps}>
      <div>{children}</div>
    </button>
  );
};

LinkButton.propTypes = {
  title: PropTypes.string.isRequired,
};

export default styled(LinkButton)`
  border: none;
  background: transparent;
  color: ${theme.palette.primary};
  font-family: inherit;
  font-size: inherit;
  font-weight: ${theme.fonts.bold};
  text-transform: uppercase;
  padding-left: 0;
  text-align: left;
  cursor: pointer;
  max-width: 100%;

  &:active {
    color: ${theme.palette.light};
  }

  &:focus {
    outline: none;
  }

  &::-moz-focus-inner {
    border: 0;
  }

  & > div {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }
`;
