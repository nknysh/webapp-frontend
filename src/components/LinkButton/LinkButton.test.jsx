import React from 'react';

import LinkButton from './';

const defaultProps = {
  title: 'My button',
};

describe('<LinkButton />', () => {
  describe('render', () => {
    it('matches snapshots (default props)', () => {
      const subject = shallow(<LinkButton {...defaultProps}>Test Button</LinkButton>);
      expect(subject).toMatchSnapshot();
    });
  });
});
