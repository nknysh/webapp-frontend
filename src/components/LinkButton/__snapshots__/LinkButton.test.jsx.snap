// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> render matches snapshots (default props) 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-hZSUBg",
        "isStatic": false,
        "rules": Array [
          "
  border: none;
  background: transparent;
  color: ",
          "#A18265",
          ";
  font-family: inherit;
  font-size: inherit;
  font-weight: ",
          "600",
          ";
  text-transform: uppercase;
  padding-left: 0;
  text-align: left;
  cursor: pointer;
  max-width: 100%;

  &:active {
    color: ",
          "#9D9591",
          ";
  }

  &:focus {
    outline: none;
  }

  &::-moz-focus-inner {
    border: 0;
  }

  & > div {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }
",
        ],
      },
      "displayName": "Styled(Component)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-hZSUBg",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  title="My button"
>
  Test Button
</StyledComponent>
`;
